content/docs/user-guide/project-structure/dvcyaml-files.md

```diff
@@ -13,6 +13,32 @@ modify, write, or generate them by your own means.
 `dvc.yaml` files are designed to be small enough so you can easily version them
 with Git along with other <abbr>DVC files</abbr> and your project's code.
 
+## Artifacts
+
+This section allows you to declare structured metadata about your artifacts.
+Although you can specify artifacts of any `type`, we are in the process of
+building a DVC-based [model registry](/doc/use-cases/model-registry) that will
+use any artifacts with `type: model`. Specically, they will appear in
+[Studio Model Registry](/doc/studio) (coming soon).
+
+```yaml
+artifacts:
+  cv-classification: # artifact ID (name)
+    path: models/resnet.pt
+    type: model
+    desc: 'CV classification model, ResNet50'
+    labels:
+      - resnet50
+      - classification
+    meta:
+      framework: pytorch
+```
+
+Artifact IDs
+[must](https://github.com/iterative/dvc/blob/main/dvc/repo/artifacts.py#L16)
```

Comment:
Is it common for our docs to refer to lines in the codebase like this?

Reply:
Not sure, but I felt like giving this link should be helpful for those who're looking for more details.

Reply:
@aguschin Do you know if using a number as the first character is problematic? If not, could we drop the restriction for `The first character must be a letter.`? The rest of the logic seems pretty straightforward.

Reply:
From the top of my head, can't think why it can be a problem. I copied your suggestion to https://github.com/iterative/dvc/pull/9345, so I can check this and update restrictions in that PR. For this docs PR, let's not change what's currently here since it's what is implemented rn.

Reply:
I also think we should avoid this - it's a bit jarring to jump into code, and also it adds fragility if we do as little as add/remove an import line and this will move 1 line down/up.
I would just copy the regex into docs and decouple it from the line in the code base and save the reader a "jump"

Reply:
ok, fixed that.